core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -441,7 +441,6 @@ class ParallelFor<FunctorType, Kokkos::TeamPolicy<Properties...>,
     // FIXME_SYCL optimize
     if (m_team_size < 0) m_team_size = 32;
 
-    // FIXME_SYCL modify for reduce etc.
     m_shmem_begin = (sizeof(double) * (m_team_size + 2));
     m_shmem_size =
         (m_policy.scratch_size(0, m_team_size) +
```

Comment:
I know we haven't been reusing code, but if we're going to have something that's duplicated but very slightly different like this (from, e.g., `Kokkos_Cuda_Parallel.hpp:790-795`), can we at least add comments talking about what's different and why?

Reply:
Are you referring to `m_shmem_size`or something else? AFAICT, its initialization looks the same for `SYCL` and `CUDA`.

Reply:
Ah, you mean `m_scratch_size[0]` which is slightly different. It turns out that this variable is not used for `Cuda` but `m_shmem_size` instead. For `SYCL` we actually use `m_scratch_size[0]` instead of `m_shmem_size` in some places (of course they still have the same value). We could probably just replace the `m_scratch_size` C-array by a simple `int` having the value of the current `m_scratch_size[1]` in the related backends.